src/__tests__/TreeViewOnNodeSelect.test.tsx

```diff
@@ -198,6 +204,89 @@ test("onNodeSelect should not be called when controlled selection", () => {
   expect(console.log).not.toBeCalled();
 });
 
+test(
+  "onNodeSelect should be called when manually node changes from not selected to selected", () => {
+    const { container } = render(
+      <TreeViewOnNodeSelect />
+    );
+    const getNodes = () => container.querySelectorAll('[role="treeitem"]');
+  
+    if (document.activeElement == null)
+      throw new Error(
+        `Expected to find an active element on the document (after focusing the second element with role["treeitem"]), but did not.`
+      );
+  
+    fireEvent.click(getNodes()[2].getElementsByClassName("checkbox-icon")[0]); //select Bananas
```

Comment:
I would suggest the use of the userEvent package in place of fireEvent.

Reply:
@kpustakhod All tests _need_ to be refactored or all tests _should_ be refactored (for consistency)?

Reply:
@mellis481 i meant _should_ be refactored (for consistency)

Reply:
@kpustakhod OK. We or someone else can do it later.

Reply:
@douglasjohnson Are you good w/ the functional changes in this PR?